overview/content-types/nix-config.nix

```
@@ -1,29 +1,77 @@
 {
-  name,
-  config,
   lib,
+  config,
+  options,
+  pkgs,
   ...
 }:
 let
   inherit (lib) mkOption types;
 in
 {
-  options = {
-    __toString = mkOption {
-      type = with types; functionTo str;
-      readOnly = true;
-      default =
-        self:
-        let
-          separator = "'$'\\n''";
-          concatSettings = lib.concatStringsSep separator (
-            lib.map (attrs: "${attrs.name} = ${toString attrs.value}") self.settings
-          );
-        in
-        "NIX_CONFIG='${concatSettings}'";
-    };
-    settings = mkOption {
-      type = with types; listOf (submodule ./config-item.nix);
+  options =
+    let
+      nix-module = import "${pkgs.path}/nixos/modules/config/nix.nix" {
+        inherit config lib pkgs;
+      };
+    in
+    {
+      inherit (nix-module.options) nix;
+
+      __toString = mkOption {
+        type = with types; functionTo str;
+        readOnly = true;
+        default =
+          self:
+          with lib;
+          let
+            mkValueString =
+              v:
+              if v == null then
+                ""
+              else if isInt v then
+                toString v
+              else if isBool v then
+                boolToString v
+              else if isFloat v then
+                floatToString v
+              else if isList v then
+                toString v
+              else if isDerivation v then
+                toString v
+              else if builtins.isPath v then
+                toString v
+              else if isString v then
+                v
+              else if strings.isConvertibleWithToString v then
+                toString v
+              else
+                abort "The nix conf value: ${toPretty { } v} can not be encoded";
+
+            mkKeyValue = k: v: "${escape [ "=" ] k} = ${mkValueString v}";
+
+            mkKeyValuePairs = attrs: concatStringsSep "\n" (mapAttrsToList mkKeyValue attrs);
+
+            isExtra = key: hasPrefix "extra-" key;
+          in
+          "export NIX_CONFIG='${lib.trim ''
+            ${mkKeyValuePairs (filterAttrs (key: value: !(isExtra key)) self.settings)}
+            ${mkKeyValuePairs (filterAttrs (key: value: isExtra key) self.settings)}
+          ''}'";
```

```suggestion
          lib.trim ''
            ${mkKeyValuePairs (filterAttrs (key: value: !(isExtra key)) self.settings)}
            ${mkKeyValuePairs (filterAttrs (key: value: isExtra key) self.settings)}
          '';
```

And let the `export NIX_CONFIG` be part of the shell command inside the instructions.

---

Ah, right. This makes more sense.